### PR TITLE
chore(flags, etc): add `created_from` property to `feature flag created` events 

### PR DESCRIPTION
**Pull request by dmarticus**
## Problem

Right now we use `feature flag created` events all over our growth metrics as indicators that customers are using our product.  However, given that users can create feature flags in the context of experiments, surveys, early access features, and the flag product itself, it'd be nice to have more granularity on where these events come from so that we can have better metrics.

## Changes

All backend stuff here; I modified the FeatureFlagSerializer to accept this new `created_from` field that gets passed into the `report_user_action` method as a `feature flag created` event property.  Then, I modified the various places where we use the FeatureFlagSerializer to include the `created_from` field of the relevant product.  

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Automated tests for the various different product serializers.

### CODE REVIEW

File: frontend/src/scenes/feature-flags/featureFlagLogic.ts [hunk 1]
```diff
@@ -302,7 +302,7 @@ export const featureFlagLogic = kea<featureFlagLogicType>([
     }),
     forms(({ actions, values }) => ({
         featureFlag: {
-            defaults: { ...NEW_FLAG } as FeatureFlagType,
```
Comment (dmarticus) [inline frontend/src/scenes/feature-flags/featureFlagLogic.ts:305]: didn't need this typecast

File: posthog/api/feature_flag.py [hunk 2]
```diff
@@ -115,6 +115,14 @@ class FeatureFlagSerializer(TaggedItemSerializerMixin, serializers.HyperlinkedMo
     )
     can_edit = serializers.SerializerMethodField()
 
+    CREATED_FROM_CHOICES = ("flags", "experiments", "surveys", "early_access_features", "web_experiments")
```
Comment (havenbarnes) [inline posthog/api/feature_flag.py:118]: nit: do we have a convention for referring to us as "flags" vs "feature_flags" that we should keep consistent?

**Review by havenbarnes — APPROVED** [review]

File: posthog/api/feature_flag.py (hunk 2 above)
Comment (dmarticus) [reply posthog/api/feature_flag.py:118]: hmmm gp gp gp.  I'll switch it to `feature_flags`

File: posthog/api/early_access_feature.py [hunk 3]
```diff
@@ -203,6 +203,7 @@ def create(self, validated_data):
                     "key": feature_flag_key,
                     "name": f"Feature Flag for Feature {validated_data['name']}",
                     "filters": filters,
+                    "created_from": "early_access_features",
```
Comment (raquelmsmith) [inline posthog/api/early_access_feature.py:206]: I recently used `creation_context` elsewhere for this kind of thing... I'm not sure if that's a better property name, but might be worth being consistent. Up to you though!

**Review by raquelmsmith — APPROVED** [review]
LGTM!

File: posthog/api/early_access_feature.py (hunk 3 above)
Comment (dmarticus) [reply posthog/api/early_access_feature.py:206]: lol I literally debated this for a while before choosing "created_from" (I also consider "creation_source").  I'll use `creation_context` to be semi-consistent.

**Pull request merged**